napari/plugins/_hookexec.py

```diff
@@ -0,0 +1,131 @@
+"""Modified _hookexec caller.
+
+TODO: TEMPORARY WORKAROUND! remove if/when pluggy merges #253
+The code here was submitted to pluggy as a PR:
+https://github.com/pytest-dev/pluggy/pull/253
+
+This module is here as a patch because pluggy doesn't indicate WHICH hook
+implementation returned the result(s)... making it hard to provide useful
+tracebacks and feedback if one of them errors.
+
+Example
+-------
+In addition to calling hooks the normal way:
+>>> plugin_manager.hook.my_hook(arg=arg)
+
+you can use _hookexec from this module as follows:
+>>> (result, imp) = _hookexec(
+...                     plugin_manager.hook.my_hook,
+...                     arg=arg,
+...                     with_impl=True,
+...                 )
+"""
+import sys
+from typing import Optional, Sequence, Any
+
+from pluggy.callers import HookCallError, _raise_wrapfail, _Result
+from pluggy.hooks import HookImpl
+from .exceptions import PluginError
+
+
+# Vendored with slight modifications from pluggy:
+# https://github.com/pytest-dev/pluggy/blob/master/src/pluggy/callers.py#L157
+def _multicall(hook_impls, caller_kwargs, firstresult=False):
+    """Execute a call into multiple python functions/methods and return the
+    result(s).
+
+    ``caller_kwargs`` comes from _HookCaller.__call__().
+    If ``caller_kwargs`` contains a key ``with_impl`` that evaluates to true,
+    results will be returned as 2-tuples of (result, hook_impl) instead of the
+    bare result.
```

Comment:
I know this is vendored but it would definitely benefit from a more complete, SciPy-style docstring, including with examples. Perhaps some example uses of `_multicall` could go in the above module docstring?

Reply:
Sure.  With this last modification (wrapping `res = hook_impl.function(*args)` in a try/catch), I'm actually starting to think that we may well need to keep a fair amount of permanent vendored/modified pluggy code.  I'm not sure we'll be able to expect pluggy to integrate all the changes we want... so I'll just remove the "temporary" comments and flesh out our internal comments